project/chapters/models.py

```diff
@@ -17,22 +24,80 @@ class ChaptersIndexPage(Page):
     parent_page_types = [
         "home.HomePage",
     ]
-    subpage_types = [
-        "chapters.Chapter"
-    ]
+
+    subpage_types = []
 
     max_count = 1
 
+    additional_context = "groups"
+    context_order = ("title",)
+
+    class Meta:
+        abstract = True
+
     def get_context(self, request):
         """
-        return a Queryset of Chapters
+        return a Queryset of Groups
         """
         context = super().get_context(request)
-        context['chapters'] = Chapter.objects.child_of(self).live()
+        context[self.additional_context] = (
+            Chapter.objects.child_of(self).live().order_by(*self.context_order)
+        )
         return context
 
 
-class Chapter(Page):
+
+class ChaptersIndexPage(GroupsIndexPage):
+    subpage_types = ['chapters.Chapter']
+    additional_context = "chapters"
+    context_order = ("region","title")
+
+
+class Group(Page):
+    introduction = RichTextField()
+
+    content_panels = Page.content_panels + [
+        FieldPanel("introduction", classname="full"),
+    ]
+
+    members = models.ManyToManyField(User, blank=True)
+
+    class Meta:
+        abstract = True
+
+    def get_context(self, request, *args, **kwargs):
+        context = super().get_context(request, *args, **kwargs)
+        """
+        If user is a member display the greyed out button using
+        the "context.member" variable, else display blue button
+        for joining group
+        """
```

Comment:
Docstring should be the first line within the function
```suggestion
        """
        If user is a member display the greyed out button using
        the "context.member" variable, else display blue button
        for joining group
        """
        
        context = super().get_context(request, *args, **kwargs)
```